infoblox/resource_zone_forward.go

```diff
@@ -0,0 +1,378 @@
+				Description:  "The name of this DNS zone. For a reverse zone, this is in “address/cidr” format",
+				Required:     false,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: checkLeadingTrailingSpaces,
+			},
+			"locked": {
+				Type:        schema.TypeBool,
+				Description: "If you enable this flag, other administrators cannot make conflicting changes. This is for administration purposes only. The zone will continue to serve DNS data even when it is locked.The default value is False.",
+				Optional:    true,
+				Default:     false,
+			},
+			"locked_by": {
+				Type:         schema.TypeString,
+				Description:  "The name of a superuser or the administrator who locked this zone.Not writable",
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: checkLeadingTrailingSpaces,
+			},
+			"mask_prefix": {
+				Type:         schema.TypeString,
+				Description:  "IPv4 Netmask or IPv6 prefix for this zone.Not Writable",
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: checkLeadingTrailingSpaces,
+			},
+			"ns_group": {
+				Type:         schema.TypeString,
+				Description:  "A forwarding member name server group. Values with leading or trailing white space are not valid for this field. The default value is undefined.",
+				Optional:     true,
+				ValidateFunc: checkLeadingTrailingSpaces,
+			},
+			"parent": {
+				Type:         schema.TypeString,
+				Description:  "The parent zone of this zone. Note that when searching for reverse zones, the “in-addr.arpa” notation should be used. Not writable.",
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: checkLeadingTrailingSpaces,
+			},
+			"prefix": {
+				Type:         schema.TypeString,
+				Description:  "The RFC2317 prefix value of this DNS zone.",
+				Optional:     true,
+				ValidateFunc: checkLeadingTrailingSpaces,
+			},
+			"using_srg_associations": {
+				Type:        schema.TypeBool,
+				Description: "This is true if the zone is associated with a shared record group. Not writable",
+				Optional:    true,
+				Computed:    true,
+			},
+			"view": {
+				Type:         schema.TypeString,
+				Description:  "The name of the DNS view in which the zone resides",
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: checkLeadingTrailingSpaces,
+			},
+			"zone_format": {
+				Type:         schema.TypeString,
+				Description:  "Determines the format of this zone - API default FORWARD. Cannot be updated.",
+				ValidateFunc: util.ValidateZoneFormat,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+			},
+		},
+	}
+}
+
+func checkLeadingTrailingSpaces(v interface{}, k string) (ws []string, errors []error) {
+	stringToCheck := v.(string)
+	trimedString := strings.Trim(stringToCheck, " ")
+	if trimedString != stringToCheck {
+		errors = append(errors, fmt.Errorf("%q must not contain trailing or leading white space", k))
+	}
+	return
+}
```

Comment:
Would be helpful to move this to util/validation_functions.go as it's a generic function